### PR TITLE
Add flags parser

### LINKED ISSUE #70: Implement saving of missions to json file

### LINKED ISSUE #78: Fix newline at EOF in login.json after every startup

### PR DESCRIPTION
**Pull request by peironggg**
Fixes #70 
Fixes #78

### CODE REVIEW

**Comment by ngzhenteng** [comment]
LGTM! thank you

File: src/main/java/seedu/address/model/flag/Flag.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.flag;
+
+import static seedu.address.logic.parser.CliSyntax.MISSION_DEADLINE;
+import static seedu.address.logic.parser.CliSyntax.QUEST_DEADLINE;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+public class Flag {
```
Comment (ngzhenteng) [inline src/main/java/seedu/address/model/flag/Flag.java:13]: Maybe we could add a javadoc comment as follows:
Represents the string consisting of a dash followed by an alphabet such as "-m" included in the user's command.

**Review by ngzhenteng — COMMENTED** [review]
Apart from the comment I made, LGTM! Thank you

File: src/main/java/seedu/address/logic/parser/ViewCommandParser.java [hunk 2]
```diff
@@ -20,14 +21,16 @@ public ViewCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_USAGE));
         }
 
+        Flag commandFlag = ParserUtil.parseFlag(trimmedArgs);
+
         String[] nameKeywords = trimmedArgs.split("\\s+");
         String viewCommandOption = nameKeywords[0];
```
Comment (wilinetan) [inline src/main/java/seedu/address/logic/parser/ViewCommandParser.java]: I think can remove this line:
String viewCommandOption = nameKeywords[0];

**Review by wilinetan — COMMENTED** [review]
other than the comment i made the rest looks good!

**Pull request merged**